lib/controllers/base.js

```diff
@@ -55,18 +56,21 @@ Controller.prototype.route = function() {
 
   // NOTE: is there a better place to do this mapping?
   var pathOptions = endpoint.string;
-  if (app.name === 'restify') {
-    if (self.method === 'delete')
+  if (this.isRestify) {
+    if (self.method === 'delete') {
```

Comment:
I kind of just think it would be easier to have an internal method heuristically determine if its restify or not. For instance:
 - `app.name === 'restify'` => restify
 - `app.hasOwnProperty('router')` => restify
 - `app.hasOwnProperty('_router')` => express 4
 - `app.hasOwnProperty('routes')` => express 3
 - etc...